src/coreclr/jit/compiler.cpp

```diff
@@ -4829,19 +4829,19 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
         //
         DoPhase(this, PHASE_CANONICALIZE_ENTRY, &Compiler::fgCanonicalizeFirstBB);
 
-        // Compute reachability sets and dominators.
+        // Compute DFS tree and remove all unreachable blocks.
         //
-        DoPhase(this, PHASE_COMPUTE_REACHABILITY, &Compiler::fgComputeReachability);
-
-        // Scale block weights and mark run rarely blocks.
-        //
-        DoPhase(this, PHASE_SET_BLOCK_WEIGHTS, &Compiler::optSetBlockWeights);
+        DoPhase(this, PHASE_DFS_BLOCKS, &Compiler::fgDfsBlocksAndRemove);
```

Comment:
It might be better to create a `PHASE_DFS_BLOCKS2` to separate the timing and other info associated with the second invocation of the phase.

Reply:
Added